src/client/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -93,7 +94,7 @@ internal class AuthenticationRequestParameters
         public ICacheSessionManager CacheSessionManager { get; }
         public HashSet<string> Scope { get; }
 
-        public bool HasScopes => Scope != null && Scope.Count > 0;
+        public bool HasScopes => Scope != null && Scope.Count > 0 && Scope.Any(s => s != string.Empty);
```

Comment:
- This still misses a case when scope is whitespace.
- Is this in a hot path? Just thinking so that Linq doesn't regress the perf then.
- Hmm, what about changing the logic when we add to `Scope` so we don't add empty scopes or whitespace; so we can forgo this check?

Reply:
Fixed. I removed the `HasScopes` property, although it wasn't used in the hot path.